src/gc.c

```diff
@@ -3170,6 +3170,20 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
         }
     }
 
+    // Readjust the max_total_memory based on what is currently available.
+    // LLVM or C may have allocated some of the memory since the last time
+    // we checked.
+
+    uint64_t total_mem = uv_get_total_memory();
+    uint64_t constrained_mem = uv_get_constrained_memory();
```

Comment:
This isn't the calculation you want. This only checks what the OS imposes. Instead, we should look at the maxrss of the current process and compare that to the heap size we know about and use that to make an informed decision about how much memory is being stolen by non-managed allocators. It is of course possible for the OS configuration to change over the lifetime of a process, but it's somewhat unlikely and certainly not the cause of observed issues.